contracts/OptimizedForceMove.sol

```diff
@@ -381,6 +352,90 @@ contract OptimizedForceMove {
         channelStorageHashes[channelId] = keccak256(abi.encode(channelStorage));
     }
 
+    struct ChannelStorageLite {
+        uint256 finalizesAt;
+        bytes32 stateHash;
+        address challengerAddress;
+        bytes32 outcomeHash;
+    }
+
+    function respondWithAlternative(
+        FixedPart memory fixedPart,
+        uint256 largestTurnNum,
+        ForceMoveApp.VariablePart[] memory variableParts,
+        uint8 isFinalCount, // how many of the states are final
+        Signature[] memory sigs,
+        uint8[] memory whoSignedWhat,
+        bytes memory channelStorageLiteBytes
+    ) public {
+        // Calculate channelId from fixed part
+        bytes32 channelId = keccak256(
+            abi.encode(fixedPart.chainId, fixedPart.participants, fixedPart.channelNonce)
+        );
+
+        // ------------
+        // REQUIREMENTS
+        // ------------
+
+        ChannelStorageLite memory channelStorageLite = abi.decode(
+            channelStorageLiteBytes,
+            (ChannelStorageLite)
+        );
+
+        // check challenge ongoing
+        require(channelStorageLite.challengerAddress != address(0), 'No challenge ongoing!');
```

Comment:
I think this might be redundant: if there is no challenge ongoing, `finalizesAt` will be `0` and the next check will fail.

Reply:
:+1:

Reply:
Addressed in 38021d1